### ISSUE TITLE
account_validity.enabled is a required field

### ISSUE THREAD

**Issue by Sorunome** [matrix-org/synapse#6740]
### Description

Using python 3.6.8 it seems that the `account_validity.enabled` is a required field.

The following causes synapse to crash on startup (max recursion depth exceeding):
```
account_validity:
  # The account validity feature is disabled by default. Uncomment the
  # following line to enable it.
  #
  #enabled: true
```

while the following works fine:
```
account_validity:
  # The account validity feature is disabled by default. Uncomment the
  # following line to enable it.
  #
  enabled: false
```

Either the generated config should be adapted or the crash fixed.

### Steps to reproduce

- Use python 3.6.8
- try to make a new homeserver
- run it without explicitly setting `account_validity.enabled` to false

### Version information

```
(.python3) sorunome@vm-1:~$ curl http://localhost:8008/_matrix/federation/v1/version
{
    "server": {
        "name": "Synapse",
        "version": "1.9.0.dev2 (b=soru/knock,ddd3584bd,dirty)"
    }
}
```
based on `develop` branch, `a17f64361c87f06c67fd7bb5a98b54dc5a2bb4fb`

**Comment by anoadragon453** [comment]
Seems like it may be failing due to:

https://github.com/matrix-org/synapse/blob/495005360cd37009818ad4214a5462c3dd0b15a6/synapse/config/emailconfig.py#L77-L79

Though I can't see why.

**Comment by anoadragon453** [comment]
Extra logging:

```
File "/home/sorunome/synapse/synapse/config/_base.py", line 101, in __getattr__
    if self.root is None:
  File "/home/sorunome/synapse/synapse/config/_base.py", line 101, in __getattr__
    if self.root is None:
  File "/home/sorunome/synapse/synapse/config/_base.py", line 101, in __getattr__
    if self.root is None:
  [Previous line repeated 323 more times]
RecursionError: maximum recursion depth exceeded while calling a Python object
```

Relevant method:

https://github.com/matrix-org/synapse/blob/f743108a94658eb1dbaf168d39874272f756a386/synapse/config/_base.py#L91-L104

**Comment by Sorunome** [comment]
It seems to also fail with Python 3.8.1

**Comment by anoadragon453** [comment]
Using Synapse v1.8.0 and earlier we run into another problem:

```
✗ synctl restart             
Starting ...
Traceback (most recent call last):
  File "/home/user/.pyenv/versions/3.6.8/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/user/.pyenv/versions/3.6.8/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/user/code/synapse/synapse/app/homeserver.py", line 630, in <module>
    main()
  File "/home/user/code/synapse/synapse/app/homeserver.py", line 625, in main
    hs = setup(sys.argv[1:])
  File "/home/user/code/synapse/synapse/app/homeserver.py", line 310, in setup
    "Synapse Homeserver", config_options
  File "/home/user/code/synapse/synapse/config/_base.py", line 617, in load_or_generate_config
    config_dict, config_dir_path=config_dir_path, data_dir_path=data_dir_path
  File "/home/user/code/synapse/synapse/config/_base.py", line 639, in parse_config_dict
    data_dir_path=data_dir_path,
  File "/home/user/code/synapse/synapse/config/_base.py", line 254, in invoke_all
    res[name] = getattr(config, func_name)(*args, **kwargs)
  File "/home/user/code/synapse/synapse/config/registration.py", line 94, in read_config
    config.get("account_validity", {}), config
  File "/home/user/code/synapse/synapse/config/registration.py", line 30, in __init__
    self.enabled = config.get("enabled", False)
AttributeError: 'NoneType' object has no attribute 'get'
error starting (exit code: 1); see above for logs
```

**Comment by anoadragon453** [comment]
This was due to a missing call to an object's superclass. Details here: https://github.com/matrix-org/synapse/pull/6747

**Comment by richvdh** [comment]
I think (hope) this was fixed by #6747
